slsReceiverSoftware/src/FrameSynchronizerApp.cpp

```diff
@@ -0,0 +1,575 @@
+// SPDX-License-Identifier: LGPL-3.0-or-other
+// Copyright (C) 2021 Contributors to the SLS Detector Package
+/* Creates the slsFrameSynchronizer for running multiple receivers in different
+ * threads form a single binary that will spit out zmq streams without
+ * reconstructing image. Sample python script for pull socket for this combiner
+ * in python/scripts folder. TODO: Not handling empty frames from one socket
+ */
+#include "sls/Receiver.h"
+#include "sls/ToString.h"
+#include "sls/container_utils.h"
+#include "sls/logger.h"
+#include "sls/sls_detector_defs.h"
+
+#include <csignal> //SIGINT
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <mutex>
+#include <ostream>
+#include <semaphore.h>
+#include <sys/socket.h>
+#include <sys/wait.h> //wait
+#include <thread>
+#include <unistd.h>
+
+#include <set>
+#include <vector>
+#include <zmq.h>
+
+std::vector<std::thread> threads;
+std::vector<sem_t *> semaphores;
+sls::TLogLevel printHeadersLevel = sls::logDEBUG;
+
+/** Define Colors to print data call back in different colors for different
+ * recievers */
+#define PRINT_IN_COLOR(c, f, ...)                                              \
+    printf("\033[%dm" f RESET, 30 + c + 1, ##__VA_ARGS__)
+
+/** Structure handling different threads */
+struct Status {
+    bool starting = true;
+    bool terminate = false;
+    unsigned long num_receivers;
+    sem_t available;
+    std::mutex mtx;
+    std::vector<zmq_msg_t *> headers;
+    std::map<unsigned int,
```

Comment:
De facto, it's the same type but I would be in favour of using uint64_t since that is what we have in the header.